packit/cli/copr_build.py

```diff
@@ -96,13 +121,20 @@ def copr_build(
         *targets.split(","), default="fedora-rawhide-x86_64"
     )
 
+    additional_repos_list: Optional[List[str]] = additional_repos.split(
+        ","
+    ) if additional_repos else None
+
     build_id, repo_url = api.run_copr_build(
         project=project,
         chroots=list(targets_to_build),
         owner=owner,
         description=description,
         instructions=instructions,
         upstream_ref=upstream_ref,
+        unlisted_on_hp=not listed_on_hp,
+        temporary=not preserve_project,
```

Comment:
Why is this flip in the meaning required? Wouldn't be better to keep the same all over the place?

Reply:
I found negative flags a bit misleading from the UX perspective.

I can add `--temporary/--no-temporary` if you want.

Reply:
The comment was for both flags.

Negative flags are indeed bad UX, but isn't it also misleading that flags are called on way on the CLI and then their meaning inverted in the code. Wouldn't it be better to keep the name of the flags everywhere in Packit code as it's in the CLI (more intuitive for the one who reads the code), and invert the meaning just before using the value, i.e. when calling the Copr API.

Reply:
> The comment was for both flags.

:+1:

> Wouldn't it be better to keep the name of the flags everywhere in Packit code as it's in the CLI (more intuitive for the one who reads the code), and invert the meaning just before using the value, i.e. when calling the Copr API.

Can be. What do you think about dual naming for flags? (click supports that.)

- `--temporary/--no-temporary`
- `--list-on-homepage/--not-listed-on-homepage` (Or whatever we choose.)

Reply:
(*sigh* communication is hard...)

I think the flags are okay the way they are on the CLI (except the `--list-whatever`, but that's another discussion).

My issue is this:

As a developer I see that there is a flag called `--preserve-project`/`--list-on-homepage`. So I'll keep looking for a boolean variable (flag) in the code, called `preserve_project`/`list_on_homepage`. But this will give only a limited number of matches, because the flag is immediately renamed to `temporary`/`unlisted_whatever` when passed to the API (which also creates a mismatch between the CLI and the API).

I think this makes the code more difficult to understand and follow. One way to solve this, is to keep the naming consistent and make the flip in the meaning only at the point when the value is passed to the Copr API, that is in Packit code base there would be no boolean variable (flag) called `temporary`/`unlisted-whatever`, only the keyword arguments of the Copr API would be called like that.

In short: the naming of these flags would be consistent in Packit from the CLI to the API all the way down to the actual call to Copr API, and when reading the code one wouldn't have to remember that "ah, this flag is called `--preserve-project` on the CLI but `temporary` when calling the API.

Reply:
OK, that makes sense. I'll change all the variables (and config) to use these:

- `preserve_project` (defaults to `False`, CLI option `--preserve-project`)
- `list_on_homepage` (defaults to `False`, CLI option `--list-on-homepage`) (I know it's not ideal, but I would like to pick one and move forward.)

And I'll switch them during the copr calls.

Will that work for you?

Reply:
> Will that work for you?

Sure, thank you :slightly_smiling_face: